### ISSUE TITLE
C++ overloaded operator parsing

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#861]
Sphinx has trouble parsing the following overloaded operators in C++:

```
#!sphinx
.. cpp:function:: MyClass & MyClass::operator++()



#!sphinx
.. cpp:function:: MyClass::pointer MyClass::operator->()
```

These two functions produce:  WARNING: Invalid definition: expected parentheses for function 

The warnings are produced due to the '->' and '++' characters

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/861
- Originally reported by: Anonymous
- Originally created at: 2012-01-25T04:49:28.954

**Comment by shimizukawa** [comment]
_From Armin Ronacher on 2012-01-25 12:41:22+00:00_

Fixed a priority error that caused operator++ and operator-> to not be parsed properly in the C++ domain. This fixes #861

→ <<cset 8bac40f75204100d4b6a4a9dc0dfc91abf8db8fb>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-01-29 08:51:29+00:00_

Fixed a priority error that caused operator++ and operator-> to not be parsed properly in the C++ domain. This fixes #861

→ <<cset 8bac40f75204100d4b6a4a9dc0dfc91abf8db8fb>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:32:54+00:00_

Removing version: 1.1.1 (automated comment)
